plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.ts

```diff
@@ -0,0 +1,313 @@
+            taskId,
+            taskInstanceId: uuid.v4(),
+          });
+          try {
+            await this.refresh(logger);
+          } catch (error) {
+            logger.error(error);
+          }
+        },
+      });
+    };
+  }
+
+  async refresh(logger: Logger) {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    const gitHubConfig = this.integrations.github.byUrl(
+      this.options.orgUrl,
+    )?.config;
+
+    if (!gitHubConfig) {
+      throw new Error(
+        `There is no GitHub integration that matches ${this.options.orgUrl}. Please add a configuration entry for it under integrations.github`,
+      );
+    }
+
+    const { org, repoSearchPath } = parseUrl(this.options.orgUrl);
+
+    const { headers } = await this.githubCredentialsProvider.getCredentials({
+      url: this.options.orgUrl,
+    });
+
+    const client = graphql.defaults({
+      baseUrl: gitHubConfig.apiBaseUrl,
+      headers,
+    });
+
+    const startTimestamp = Date.now();
+
+    logger.info(`Reading GitHub repositories from ${this.options.orgUrl}`);
+
+    if (!this.options.files) {
+      throw new Error(
+        `Missing 'files' array key in 'options' passed to GithubEntityProvider.fromConfig() : ${this.getProviderName()}`,
+      );
+    }
+
+    const expectedEntityFiles = this.options.files;
+
+    const { repositories } = await getOrganizationEntities(
+      client,
+      org,
+      expectedEntityFiles,
+    );
+
+    const matching = repositories.filter(
+      r =>
+        !r.isArchived &&
+        repoSearchPath.test(r.name) &&
+        r.defaultBranchRef?.name,
+    );
+
+    const duration = ((Date.now() - startTimestamp) / 1000).toFixed(1);
+
+    logger.info(
+      `Read ${repositories.length} GitHub repositories (${matching.length} matching the pattern) in ${duration} seconds`,
+    );
+
+    const files = expectedEntityFiles.map(fileName => ({
+      name: fileName,
+      key: camelCase(fileName),
+    }));
+
+    const repoEntities: Array<LocationSpec | Array<LocationSpec>> = [];
+
+    for (const entity of matching) {
+      for (const { name: fileName, key } of files) {
+        const entityObj: any = entity[key as keyof typeof entity];
+        if (entityObj) {
+          const locationEntity = this.createLocationSpec({
+            url: entity.url,
+            branch: entity.defaultBranchRef?.name || '',
+            file: fileName,
+          });
+          repoEntities.push(locationEntity);
+        }
+      }
+    }
+
+    const entityLocations = repoEntities.flat();
+
+    if (entityLocations.length === 0) {
+      logger.info(`No valid entities found by ${this.getProviderName()}.`);
+      return false;
+    }
+
+    await this.connection.applyMutation({
+      type: 'full',
+      entities: entityLocations.map(location => ({
+        locationKey: this.getProviderName(),
+        entity: locationSpecToLocationEntity({ location }),
+      })),
+    });
+
+    logger.info(
```

Comment:
you wanted this line before the `applyMutation` right?

Reply:
resolved